src/main/java/org/rumbledb/runtime/functions/sequences/general/SubsequenceFunctionIterator.java

```diff
@@ -50,107 +50,107 @@ public SubsequenceFunctionIterator(
             ExceptionMetadata iteratorMetadata
     ) {
         super(parameters, executionMode, iteratorMetadata);
-    }
-
-    @Override
-    public void open(DynamicContext context) {
-        super.open(context);
+        this.sequenceIterator = this.children.get(0);
 
-        this.currentPosition = 1; // JSONiq indices start from 1
+        Item positionItem = this.children.get(1)
```

Comment:
The constructor of an iterator cannot manipulate children iterators -- this must be done in open() (for local), and in getRDDAux() (for parallel). This implementation is likely to break as soon as you provide a variable for parameters 2 and 3.

Reply:
I see. This should now be fixed.